Doc/library/socket.rst

```diff
@@ -595,6 +595,26 @@ The following functions all create :ref:`socket objects <socket-objects>`.
    .. versionchanged:: 3.2
       *source_address* was added.
 
+.. function:: create_server_sock(address, *, dual_stack=True, queue_size=5, reuse_addr=True)
+
+   Convenience function which creates a TCP server bound to *address* (a
+   2-tuple ``(host, port)``) and return a socket object.  If *host* is an empty
+   string or ``None`` all network interfaces are used.
+   If dual-stack is supported by kernel and *dual_stack* parameter
+   is ``True`` the socket will be able to serve both :data:`AF_INET` and
+   :data:`AF_INET6` (IPv4 / IPv6) connections. If not the right address family
+   will be chosen based on the host specified in *address*.
+   *queue_size* is the argument passed to :meth:`socket.listen`.
+   If *reuse_addr* is ``True`` :data:`socket.SO_REUSEADDR` socket flag is set.
+
+   .. versionadded:: 3.8
+
+.. function:: has_dual_stack()
+
+   Return ``True`` if the kernel supports creating a socket which can handle
+   both :data:`AF_INET` and :data:`AF_INET6` (IPv4 / IPv6) connections.
```

Comment:
I think the name of this function is pretty confusing. A "dual stack host" is one that has both IPv4 and IPv6 support. A host can have dual stack support without being able to support both of them in a single socket.

Also, for Python docs it should probably be "the platform", not "the kernel".

Reply:
I agree the name is not the best. Actually, on a second thought, I don't think it's necessary to expose this function after all.